Sources/SwiftProtobuf/JSONScanner.swift

```diff
@@ -190,8 +190,9 @@ private func parseBytes(
     // a closing double quote.
     index = digitsStart
     try value.withUnsafeMutableBytes {
-        (dataPointer: UnsafeMutablePointer<UInt8>) in
-        var p = dataPointer
+        (body: UnsafeMutableRawBufferPointer) in
+      if let baseAddress = body.baseAddress, body.count > 0 {
```

Comment:
The `value` here is actually the output buffer; the code here assumes that we're going to scan the input even if the output is zero-sized.  In particular, the close quote gets consumed deep inside the scan loop.  Since the code now skips that loop for zero-byte output, we end up not consuming the close quote for `""`.

I agree this is a somewhat confusing design.  I think the best fix would change how we handle the close quote:
* Look for two `case asciiDoubleQuote` below and remove `source.formIndex(after: &index)` at those points.
* Add `source.formIndex(after: &index)` just before `return value` at the end of the function.
This moves the close-quote handling outside of the `withUnsafeMutableBytes` which ensures that we skip it even if the output is empty.

Reply:
Ah, makes total sense. Making those changes now. Thanks!